arch/arm/boot/dts/bcm2838.dtsi

```diff
@@ -222,15 +223,12 @@
 	};
 
 	arm-pmu {
-		/*
-		 * N.B. the A72 PMU support only exists in arch/arm64, hence
-		 * the fallback to the A53 version.
-		 */
-		compatible = "arm,cortex-a72-pmu", "arm,cortex-a53-pmu";
+		compatible = "arm,cortex-a72-pmu";
```

Comment:
Marc said:
> You're thus guaranteed to count the wrong events... And you could at
least have "arm,armv8-pmuv3".

Please explain the effect of your change (or why we've ended up with this) in a comment.

Reply:
There's no such thing as fall-back from a72 to a53 - the A53 has a unique event type of ARMV8_A53_PERFCTR_PREF_LINEFILL, and all other event types supported by A53 are a subset of events supported by A72.

The the arm64/kernel/perf_event.c driver handles various pmuv3 implementations including the A72 variant. Matching on the generic "arm,armv8-pmuv3" string does not enable optional implementation-dependent event types, of which the A72 has several that are likely to be useful.

Also, there are dts files in the kernel tree that use both the a53 and generic compatible strings, but there are no dts files with more than just the a72 compatible string.